birdhouse/components/weaver/default.env

```diff
@@ -0,0 +1,59 @@
+# ------
+# weaver
+# ------
+
+# add any new variables not already in 'VARS' or 'OPTIONAL_VARS' that must be replaced in templates here
+# single quotes are important in below list to keep variable names intact until 'pavics-compose' parses them
+EXTRA_VARS='
+  $WEAVER_CONFIG
+  $WEAVER_MANAGER_NAME
+  $WEAVER_WORKER_NAME
+  $WEAVER_WPS_NAME
+  $WEAVER_WPS_OUTPUTS_DIR
+  $WEAVER_WPS_OUTPUTS_PATH
+  $WEAVER_WPS_PATH
+  $WEAVER_WPS_WORKDIR
+  $WEAVER_MANAGER_LOG_LEVEL
+  $WEAVER_WORKER_LOG_LEVEL
+'
+# extend the original 'VARS' from 'birdhouse/pavics-compose.sh' to employ them for template substitution
+# adding them to 'VARS', they will also be validated in case of override of 'default.env' using 'env.local'
+VARS="$VARS $EXTRA_VARS"
```

Comment:
You sure you want to add to `VARS` and not `OPTIONAL_VARS`?  I have not checked but if all these have default values, `OPTIONAL_VARS` is much more appropriate.  See https://github.com/bird-house/birdhouse-deploy/blob/e10a2ef9bd77498f1af22267cad5d181d0da65ef/birdhouse/components/monitoring/default.env for example

Reply:
All variables are mandatory in the sense that if they are overridden by an empty string, compose won't work at all. 
So although they are *optional* in terms of "no need any override", they must be validated. 
My understanding of `OPTIONAL_VARS` was that empty value will still pass. 
What I wrong? If so, I don't understand why there are 2 variables.